File: лаба2/Go/laba2_1zadanie.go
package main

import (
	"fmt"
	"math"
)

// Проверяет, является ли число простым
func isprostoe(N int) bool {
	for i := 2; i <= int(math.Sqrt(float64(N))); i++ {
		if N%i == 0 {
			return false
		}
	}
	return N > 1 // 1 не простое
}

// Находит ближайшее простое число к N
func near(N int) int {
	if isprostoe(N) {
		return N
	}
	lower := N - 1
	upper := N + 1

	for {
		if isprostoe(lower) {
			return lower // ближайшее простое снизу
		}
		if isprostoe(upper) {
			return upper // ближайшее простое сверху
		}
		lower--
		upper++
	}
}

func main() {
	fmt.Print("Пример: ")
	var N int
	fmt.Scan(&N) // Ввод числа

	num := near(N)      // Поиск ближайшего простого
	res := N - num      // Разница

	fmt.Printf("Ближайшее простое: %d\n", num)
	fmt.Printf("Результат: %d\n", res)
}

File: лаба2/Kotlin/laba2_1zadanie.kt
import kotlin.math.sqrt

// Функция проверки, является ли число простым
fun isprostoe(N: Int): Boolean {
    if (N <= 1) return false  // числа <= 1 не считаются простыми

    // Проверяем делимость от 2 до квадратного корня из N
    for (i in 2..sqrt(N.toDouble()).toInt()) {
        if (N % i == 0) return false  // если делится без остатка - не простое
    }
    return true  // если делителей нет - число простое
}

// Функция поиска ближайшего простого числа к N
fun near(N: Int): Int {
    if (isprostoe(N)) return N  // если N простое - возвращаем его

    var lower = N - 1  // начинаем искать вниз
    var upper = N + 1  // и вверх

    while(true) {
        if (isprostoe(lower)) return lower  // нашли простое снизу
        if (isprostoe(upper)) return upper  // нашли простое сверху
        lower--
        upper++
    }
}

fun main() {
    val N = readLine()?.toIntOrNull() ?: 0  // читаем число из ввода, если пусто - 0
    val num = near(N)                       // ищем ближайшее простое
    val res = N - num                       // вычисляем разницу

    println("Пример: $N")
    println("Ближайшее простое число: $num")
    println("Результат: $res")
}

